src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/EditAppealAfterSubmitHandler.java

```diff
@@ -107,24 +117,38 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
                         parse(maybeHomeOfficeDecisionDate
                                 .orElseThrow(() -> new RequiredFieldMissingException("homeOfficeDecisionDate is missing")));
             }
+
+            if (decisionDate != null
+                && decisionDate.isBefore(dateProvider.now().minusDays(maybeOutOfCountryDecisionType.isPresent() ? appealOutOfTimeDaysOoc : appealOutOfTimeDaysUk))) {
+                asylumCase.write(SUBMISSION_OUT_OF_TIME, YES);
+            } else {
+                asylumCase.write(SUBMISSION_OUT_OF_TIME, NO);
+                asylumCase.clear(APPLICATION_OUT_OF_TIME_EXPLANATION);
+                asylumCase.clear(APPLICATION_OUT_OF_TIME_DOCUMENT);
+                asylumCase.clear(RECORDED_OUT_OF_TIME_DECISION);
+            }
+
         } else {
             //do nothing
```

Comment:
Can we please change the comment - to explain what we are doing for ADA appeals.

Reply:
super spot @neha-aggarw   - I'll remove it as it was a point in time thing